os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_ota_ram.c

```diff
@@ -75,4 +75,121 @@ u32 OTA_Change(u32 OTAIdx)
 	return _TRUE;
 }
 
+#define TIZENRT_KERNEL_HEADER_LEN 0x10
+IMAGE2_RAM_TEXT_SECTION
+static u8 OTA_AddrChk(u8 idx, uint32_t pAddr, uint32_t *size, uint32_t *SBAddr)
+{
+	IMAGE_HEADER *Img2Hdr, *Img2DataHdr, *PsramHdr;
+	uint32_t ImgSize, SBheader;
+	u8 res = _TRUE;
+
+	/* Verify signature */
+	if(idx == 0) {
+		Img2Hdr = (IMAGE_HEADER *)(pAddr + TIZENRT_KERNEL_HEADER_LEN);
+	} else {
+		Img2Hdr = (IMAGE_HEADER *)pAddr;
+	}
+	if((Img2Hdr->signature[0] != 0x35393138) || (Img2Hdr->signature[1] != 0x31313738) || ((Img2Hdr->image_size) & 0x3)) {
+		res = _FALSE;
+		goto exit;
+	}
+
+	/* Get image size, which should be 4KB-aligned */
+	if(idx == 0) {
+		Img2DataHdr = (IMAGE_HEADER *)(pAddr + TIZENRT_KERNEL_HEADER_LEN + IMAGE_HEADER_LEN + Img2Hdr->image_size);
+		ImgSize = Img2Hdr->image_size + Img2DataHdr->image_size + IMAGE_HEADER_LEN * 2 + TIZENRT_KERNEL_HEADER_LEN;
+	} else {
+		Img2DataHdr = (IMAGE_HEADER *)(pAddr + IMAGE_HEADER_LEN + Img2Hdr->image_size);
+		if((Img2DataHdr->signature[0] != 0x35393138) || (Img2DataHdr->signature[1] != 0x31313738) || ((Img2DataHdr->image_size) & 0x3)) {
+			res = _FALSE;
```

Comment:
Same as above.

Reply:
Done modify.

Reply:
I don't see the requested modifications here